### PR TITLE
Install mercurial in bootstrap image

### PR DESCRIPTION
**Pull request by ixdy**
This is needed to install some third-party dependencies.

I haven't yet pushed any images.

/assign @BenTheElder @fejta

### CODE REVIEW

File: images/kubekins-e2e/Dockerfile [hunk 1]
```diff
@@ -33,9 +33,11 @@ RUN mkdir -p /go/src/k8s.io/kubernetes \
 # preinstall:
 # - graphviz package for graphing profiles
 # - bc for shell to junit
+# - mercurial for downloading third-party dependencies (git is already installed)
 # - rpm for building RPMs with Bazel
 RUN apt-get install -y bc \
     graphviz \
+    mercurial \
```
Comment (BenTheElder) [inline images/kubekins-e2e/Dockerfile]: should this be in kubekins-e2e or in bootstrap?

Comment (ixdy) [reply images/kubekins-e2e/Dockerfile]: that's more work, but you're right. :)

Comment (BenTheElder) [reply images/kubekins-e2e/Dockerfile]: time to bump all the things! we've needed a kubekins bump today anyhow for some firewall cleanup stuff

**Comment by ixdy** [comment]
should I do a bootstrap build + bump in this PR? (is there even a helper script to do that, or is it manual?)

**Comment by BenTheElder** [comment]
> should I do a bootstrap build + bump in this PR? (is there even a helper script to do that, or is it manual?)

yeah. there's no helper IIRC, but it's used in far less places. it might even be fine to just bump it in kubekins-e2e for now /shrug

**Comment by dims** [comment]
LGTM 👍 i had to install this in cloud-provider-openstack Makefile.

**Comment by ixdy** [comment]
ok, I've built and pushed a new bootstrap image, and am building a canary kubekins-e2e now. PTAL?

**Comment by ixdy** [comment]
we can probably revert 430bf367051b6d1ef4b575a4eee654bce54f8601 now, though maybe not in this PR.

File: images/bootstrap/Makefile [hunk 2]
```diff
@@ -20,8 +20,8 @@ all: build
 build:
 	bazel build //images/bootstrap/barnacle:barnacle --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64
 	cp ./../../bazel-bin/images/bootstrap/barnacle/linux_amd64_pure_stripped/barnacle ./barnacle/barnacle
-	docker build --build-arg IMAGE_ARG=$(IMG):$(TAG) -t $(IMG):$(TAG) .
-	rm ./barnacle/barnacle
+	docker build --pull --build-arg IMAGE_ARG=$(IMG):$(TAG) -t $(IMG):$(TAG) .
```
Comment (BenTheElder) [inline images/bootstrap/Makefile]: why --pull? we're using date-sha tags?

Comment (ixdy) [reply images/bootstrap/Makefile]: bootstrap is based on `debian:jessie`, so we should always pull in updates before building it.

**Comment by BenTheElder** [comment]
once the kubekins-e2e images are built we should watch for a https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-canary/ run with the new image tag in the log.

File: images/bootstrap/Makefile (hunk 2 above)
Comment (BenTheElder) [reply images/bootstrap/Makefile]: er right...
/lgtm
/hold

**Comment by ixdy** [comment]
everything is looking pretty good so far. gonna bump kubekins-e2e everywhere.

**Pull request merged**